subsys/modem/backends/modem_backend_tty.c

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2022 Trackunit Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/modem/backend/tty.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(modem_backend_tty);
+
+#include <fcntl.h>
+#include <unistd.h>
+#include <poll.h>
+#include <string.h>
+
+#define MODEM_BACKEND_TTY_THREAD_PRIO          (10)
+#define MODEM_BACKEND_TTY_THREAD_RUN_PERIOD_MS (1000)
+#define MODEM_BACKEND_TTY_THREAD_POLL_DELAY    (100)
+
+#define MODEM_BACKEND_TTY_STATE_RUN_BIT (1)
+
+static void modem_backend_tty_routine(void *p1, void *p2, void *p3)
+{
+	struct modem_backend_tty *backend = (struct modem_backend_tty *)p1;
+	struct pollfd pd;
+
+	ARG_UNUSED(p2);
+	ARG_UNUSED(p3);
+
+	pd.fd = backend->tty_fd;
+	pd.events = POLLIN;
+
+	/* Run until run flag is cleared. Check every MODEM_BACKEND_TTY_THREAD_RUN_PERIOD_MS */
+	while (atomic_test_bit(&backend->state, MODEM_BACKEND_TTY_STATE_RUN_BIT)) {
+		/* Clear events */
+		pd.revents = 0;
+
+		if (poll(&pd, 1, MODEM_BACKEND_TTY_THREAD_RUN_PERIOD_MS) < 0) {
+			LOG_ERR("Poll operation failed");
+			break;
+		}
+
+		if (pd.revents & POLLIN) {
+			modem_pipe_notify_receive_ready(&backend->pipe);
+		}
+
+		k_sleep(K_MSEC(MODEM_BACKEND_TTY_THREAD_POLL_DELAY));
+	}
+}
+
+static int modem_backend_tty_open(void *data)
+{
+	struct modem_backend_tty *backend = (struct modem_backend_tty *)data;
+
+	if (atomic_test_and_set_bit(&backend->state, MODEM_BACKEND_TTY_STATE_RUN_BIT)) {
+		return -EALREADY;
+	}
+
+	backend->tty_fd = open(backend->tty_path, (O_RDWR | O_NONBLOCK), 0644);
+
+	if (backend->tty_fd < 0) {
```

Comment:
As a general note for changes in this PR, it would improve readability if the empty line between these two lines is removed. So if there is some variable set, and we immediately check its value then those two lines are tied together and there should be no empty line between them. This improves readability of the code and helps reviewing it. Please check the PR as I see these kind of checks all over the place.

Reply:
Went over every file in the PR removing empty lines that fit the example given here :)